app/validators/income_assessment/answers_validator.rb

```diff
@@ -39,13 +41,29 @@ def frozen_income_savings_assets_complete?
     def income_payments_complete?
       return true if record.has_no_income_payments == 'yes'
 
-      record.income_payments.present? && record.income_payments.all?(&:complete?)
+      record.income_payments.for_client.present? && record.income_payments.for_client.all?(&:complete?)
+    end
+
+    def partner_income_payments_complete?
+      return true unless FeatureFlags.partner_journey.enabled?
+      return true if crime_application.income.partner_has_no_income_payments.to_s == 'yes'
+      return true unless include_partner_in_means_assessment?
```

Comment:
Can you not combine guard clause 1 and 3? 
`return true unless FeatureFlags.partner_journey.enabled? && include_partner_in_means_assessment?`